### ISSUE TITLE
[Bug]: Table default selection row doesn't work when primary key is set with server-side pagination 

### ISSUE THREAD

**Issue by kamakshibhat-appsmith** [appsmithorg/appsmith#21108]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Table default selection row doesn't work when primary key is set with server-side pagination 

### Steps To Reproduce

1. Configure a query with table which has LIMIT and OFFSET set for pagination
2. Select primary key 
3. Set any value in default selected row 
4.Change pages

### Environment

Production

### Issue video log

https://www.loom.com/share/92968e00a9bf4b84a1d03c9a6a765ee3

### Version

Cloud

**Comment by rajatagrawal** [comment]
Notes from discussion with @dilippitchika 

The default selected row feature, when primary key is enabled, is working similar to its counterpart in list widget. It is trying to select rows whose primary key column value is equal to the value entered in the input box for default selected row input box.

Question : What is the intended functionality of this feature ?
Option 1 : To select rows by index for value passed in default selected row index input field
Option 2: To select rows by value for value passed in default selected row index input field.

If Option 1,
Fix: I can remove the code for Option 2 already present.

If Option 2:
The bug in this case seems like the lack of ability for the user to deselect primary key option from the UI. If the user is able to deselect primary key column, then they won't be stuck and can be unblocked for carrying out other tasks for the table widget.

Please suggest @kamakshibhat-appsmith @dilippitchika @somangshu 

cc : @keyurparalkar

**Comment by dilippitchika** [comment]
Is there any performance value people are currently getting from the feature if primary key is selected? @rajatagrawal

**Comment by rajatagrawal** [comment]
no, there aren't any performance related concerns in both approaches.

**Comment by saiprabhu-dandanayak** [comment]
Hi **@rajatagrawal , @dilippitchika , @somangshu**  , i have raised Pr for this bug , Could You assign a reviewer .
Thanks in advance.

**Comment by rajatagrawal** [comment]
Reverted [here](https://github.com/appsmithorg/appsmith/pull/33902#issuecomment-2148721907)
